src/main/java/bridge/controller/ApplicationStatus.java

```diff
@@ -0,0 +1,28 @@
+package bridge.controller;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public enum ApplicationStatus {
+
+    QUIT("Q"), RUNNING("R");
+
+    private static final Map<String, ApplicationStatus> cachedGameCommand = new HashMap<>();
+    private final String command;
+
+    static {
+        for (ApplicationStatus applicationStatus : values()) {
+            cachedGameCommand.put(applicationStatus.command, applicationStatus);
+        }
+    }
```

Comment:
이 부분을 보며 캐싱을 하기 위한 `Map`의 선언부와 데이터 삽입 부분이 분리되어 있어 아쉽다는 생각이 들었습니다.
정-말 개인적인 취향이지만, 초기 데이터가 고정되어 있을 경우 변수 선언 시점에서 초기 데이터를 확인하거나, 추론할 수 있는 코드가 조금 더 편하더라고요.

이를 위해서 아래와 같이 변경할 수 있을 것 같습니다.

```suggestion
    private static final Map<String, ApplicationStatus> cachedGameCommand = 
            Arrays.stream(values())
                    .collect(Collectors.toMap(status -> status.command, status -> status));
```

물론 코파일럿의 힘을 빌리긴 했습니다...ㅎㅎ
하지만 역시 취향의 영역으로 생각되어 편하게 생각해 주시면 좋을 것 같습니다!

+) 참고로 `Collectors.toMap()` 메서드의 경우 내부적으로 `HashMap`을 사용하고 있기에 혹여나 `EnumMap`의 사용을 고려하실 경우에는 직접 `collect()` 메서드 내부에 파라미터들을 하나하나 넣어 주어야 합니다.
이 경우 파라미터의 개수도 많고 `collect()` 메서드 자체를 잘 이해하고 있어야 하기에 되려 번거로워 질 수도 있을 것 같습니다

Reply:
생각지도 못한 부분이었습니다!
한 번쯤 고민해봐야할 부분인 것 같아요 ㅎㅎ 좋은 말씀 감사합니다 😊
`Collectors.toMap()`에 대한 상세한 코멘트도 감사합니다 😊

Reply:
저도 초기 데이터가 함께 모여 있는 것이 더 좋다고 생각합니다.

그런데 Java 언어로 `Array` -> `Map` 변환하는 코드가 조금 지저분하기도 하고 손에 익지 않아서

급한 상황에서는 최대한 가깝게 두는 정도로 타협해도 좋을 것 같습니다.👍